From 05251e181e9c28670dbb127b181aafb3c5928b52 Mon Sep 17 00:00:00 2001
From: Tomas Glozar <tglozar@gmail.com>
Date: Tue, 18 Nov 2025 19:16:11 +0100
Subject: [PATCH 93/96] ia64: Guard crashk_{start_end} with CRASH_RESERVE

crashk_start and crashk_end depend on CONFIG_CRASH_RESERVE being set.

Only set them in arch/ia64 code when CONFIG_CRASH_RESERVE is set.

Signed-off-by: Tomas Glozar <tglozar@gmail.com>
---
 arch/ia64/kernel/efi.c   | 2 ++
 arch/ia64/kernel/setup.c | 2 ++
 arch/ia64/mm/init.c      | 2 +-
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/ia64/kernel/efi.c b/arch/ia64/kernel/efi.c
index b52d4865c3cb..eb5bdd1baa39 100644
--- a/arch/ia64/kernel/efi.c
+++ b/arch/ia64/kernel/efi.c
@@ -1270,8 +1270,10 @@ efi_initialize_iomem_resources(struct resource *code_resource,
 #ifdef CONFIG_KEXEC
                         insert_resource(res, &efi_memmap_res);
                         insert_resource(res, &boot_param_res);
+# ifdef CONFIG_CRASH_RESERVE
 			if (crashk_res.end > crashk_res.start)
 				insert_resource(res, &crashk_res);
+# endif
 #endif
 		}
 	}
diff --git a/arch/ia64/kernel/setup.c b/arch/ia64/kernel/setup.c
index 8babc91bfb45..f43bedab6649 100644
--- a/arch/ia64/kernel/setup.c
+++ b/arch/ia64/kernel/setup.c
@@ -307,8 +307,10 @@ static void __init setup_crashkernel(unsigned long total, int *n)
 			rsvd_region[*n].end =
 				(unsigned long)__va(base + size);
 			(*n)++;
+# ifdef CONFIG_CRASH_RESERVE
 			crashk_res.start = base;
 			crashk_res.end = base + size - 1;
+# endif
 		}
 	}
 	efi_memmap_res.start = ia64_boot_param->efi_memmap;
diff --git a/arch/ia64/mm/init.c b/arch/ia64/mm/init.c
index 09f16a2a6f7d..401bbe14ecbb 100644
--- a/arch/ia64/mm/init.c
+++ b/arch/ia64/mm/init.c
@@ -390,7 +390,7 @@ int __init register_active_ranges(u64 start, u64 len, int nid)
 {
 	u64 end = start + len;
 
-#ifdef CONFIG_KEXEC
+#ifdef CONFIG_CRASH_RESERVE
 	if (start > crashk_res.start && start < crashk_res.end)
 		start = crashk_res.end;
 	if (end > crashk_res.start && end < crashk_res.end)
-- 
2.25.1

